src/app/util/ember-compatibility-functions.cpp

```diff
@@ -283,6 +283,15 @@ CHIP_ERROR ReadSingleClusterData(const SubjectDescriptor & aSubjectDescriptor, b
 
     const EmberAfCluster * attributeCluster            = nullptr;
     const EmberAfAttributeMetadata * attributeMetadata = nullptr;
+
+#if defined(CHIP_DISABLE_UNIQUE_ID) && CHIP_DISABLE_UNIQUE_ID
+    if (aPath.mEndpointId == 0 && aPath.mClusterId == chip::app::Clusters::BasicInformation::Id &&
+        aPath.mAttributeId == chip::app::Clusters::BasicInformation::Attributes::UniqueID::Id)
+    {
+        return CHIP_ERROR_IM_GLOBAL_STATUS_VALUE(UnsupportedAttributeStatus(aPath));
```

Comment:
What about wildcard expansion and other things that look at attribute metadata?  This will lead to incorrect behavior on _write_ attempts to this attribute, for example.

The right way to do this is to have an example app configured with this attribute turned off or something, not by hacking core SDK code like this.